MyFitness/MyFitness/Custom Views/ImageViews/NetworkManager.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  NetworkManager.swift
+//  MyFitness
+//
+//  Created by John Kouris on 3/2/20.
+//  Copyright © 2020 John Kouris. All rights reserved.
+//
+
+import UIKit
+
+class NetworkManager {
+    static let shared = NetworkManager()
+    private let baseURL = "https://api.spoonacular.com/recipes/random/"
+    private let apiKey = APIKeys.recipeApiKey
+    let cache = NSCache<NSString, UIImage>()
+    
+    private init() {}
+    
+    func getRecipes(completion: @escaping (Result<RecipeList, MFError>) -> Void) {
+        let queryItems = [URLQueryItem(name: "apiKey", value: apiKey), URLQueryItem(name: "number", value: "30")]
+        var urlComponent = URLComponents(string: baseURL)!
+        urlComponent.queryItems = queryItems
+        
+        guard let url = urlComponent.url else {
+            completion(.failure(.invalidResponse))
+            return
+        }
+        
+        let task = URLSession.shared.dataTask(with: url) { (data, response, error) in
+            if let _ = error {
+                completion(.failure(.unableToComplete))
```

Comment:
I would wrap this in a DispatchQueue.main.async call, so you don’t have unexpected threading issues later on.